### DISCUSSION TITLE
GitHub Desktop error: "The remote disconnected. Check your internet connection and try again." Bug?

### DISCUSSION THREAD

**Discussion by ar-ibrahim** [community/community#138617] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi! I wasn't able to push my commits. I was able to push some commits after updating my GitHub Desktop Version but not all. So was wondering if there is a bug? I checked the file sizes, permissions, authentication etc. and the issue was still not fixed. I've uploaded the log file if that helps! Any help would be appreciated!

[2024-09-14.desktop.production.log](https://github.com/user-attachments/files/17001456/2024-09-14.desktop.production.log)

**Comment by Sin-cy** [comment, marked as answer]
Hi. It might be something to do with the pushing size. 
Pushing from the command line should solve this.

So from your log files 
`error: RPC failed; HTTP 500 curl 22 The requested URL returned error: 500
send-pack: unexpected disconnect while reading sideband packet` 
This seems to be the issue. The File size that your are trying to push might be too big. I think you might have multiple commits you haven't pushed and you are pushing them all at once and github is rejecting it. 

Say you are trying to push to the main remote branch.
Do it from the shell / use command line by specifying the latest commit hash into the git push command 

To get commit hash run 
`git log --oneline` 

you will see a 7 digit random character infront of your latest commit ,the top most one ( ex hash : 6dhf84s )
so then you would go ahead and copy that , then run the command 
`git push origin 6dhf84s:main` 
this will push all your commits you made right until the latest one which I am assuming as 6dhf84s. You can choose your range from there.

Hope this helps.

**Reply by ar-ibrahim** [reply]
Hi! Thank you so much for your reply. You were right. It was a size problem. I got the same error message when I used the command line. But it finally worked by 'increasing the buffer size' using this command `git config --global http.postBuffer 524288000`.

**Reply by Sin-cy** [reply]
Awesome, glad it helped !!!
